core-backup/src/main/scala/io/aiven/guardian/kafka/backup/BackupClientInterface.scala

```diff
@@ -66,23 +71,60 @@ trait BackupClientInterface[T <: KafkaClientInterface] {
     */
   def empty: () => Future[BackupResult]
 
-  @nowarn("msg=not.*?exhaustive")
   private[backup] def calculateBackupStreamPositions(
       sourceWithPeriods: SourceWithContext[(ReducedConsumerRecord, Long),
                                            kafkaClientInterface.CursorContext,
                                            kafkaClientInterface.Control
       ]
-  ): SourceWithContext[(ReducedConsumerRecord, BackupStreamPosition),
-                       kafkaClientInterface.CursorContext,
-                       kafkaClientInterface.Control
-  ] = sourceWithPeriods
-    .sliding(2)
-    .map { case Seq((beforeReducedConsumerRecord, beforeDivisions), (_, afterDivisions)) =>
-      val backupStreamPosition = splitAtBoundaryCondition(beforeDivisions, afterDivisions)
+  ): Source[RecordElement, kafkaClientInterface.Control] =
+    sourceWithPeriods.asSource
+      .prefixAndTail(2)
+      // This algorithm only works with Source's that have 2 or more elements
+      .flatMapConcat {
+        case (Seq(
+                ((firstReducedConsumerRecord, firstDivision), firstContext),
+                ((secondReducedConsumerRecord, secondDivision), secondContext)
+              ),
+              rest
+            ) =>
+          val all = Source
+            .combine(
+              Source(
+                List(
+                  ((firstReducedConsumerRecord, firstDivision), firstContext),
+                  ((secondReducedConsumerRecord, secondDivision), secondContext)
```

Comment:
Nitpick, but why `Seq` a couple of lines above and `List` here?
Also, wouldn't make sense to save the `Seq` in line 84 in a variable and reuse it here? You'll save some allocations.

Reply:
This is basically due to subtyping, when we do are comparing the type of something we use `Seq` since its will match against all collection types i.e. `Vector`/`List`/`Seq` etc etc. When we manually create a collection we use `List` since thats the simplest one, but in some cases various methods on akka-streams will return a `Vector` and so using `Seq` specifically in `case` statements means it will match against whatever collection we are dealing with.